exercises/concept/ticket-please/tests/Tests.elm

```diff
@@ -0,0 +1,277 @@
+                Ticket
+                    { status = New
+                    , createdBy = ( User "John", 1 )
+                    , assignedTo = Nothing
+                    , comments = [ ( User "John", "I can't connect to the wifi." ) ]
+                    }
+                    |> numberOfCreatorComments
+                    |> Expect.equal 1
+        , test "numberOfCreatorComments with several comments" <|
+            \() ->
+                Ticket
+                    { status = InProgress
+                    , createdBy = ( User "Steve", 1 )
+                    , assignedTo = Just (User "Roy")
+                    , comments =
+                        [ ( User "Roy", "Hello, IT." )
+                        , ( User "Steve", "My computer refuses to print anything." )
+                        , ( User "Roy", "Have you tried turning it off and on again?" )
+                        , ( User "Steve", "Is that a reference to something?" )
+                        ]
+                    }
+                    |> numberOfCreatorComments
+                    |> Expect.equal 2
+        ]
+
+
+task3 : Test
+task3 =
+    describe "Task 3: TicketsPlease.assignedToDevTeam"
+        [ test "assignedToDevTeam with no one assigned" <|
+            \() ->
+                Ticket { status = New, createdBy = ( User "John", 1 ), assignedTo = Nothing, comments = [] }
+                    |> assignedToDevTeam
+                    |> Expect.false "Expected unassigned ticket"
+        , test "assignedToDevTeam with ticket assigned to non-dev team" <|
+            \() ->
+                Ticket { status = New, createdBy = ( User "Jack", 2 ), assignedTo = Just (User "Roy"), comments = [] }
+                    |> assignedToDevTeam
+                    |> Expect.false "Expected ticket not assigned to dev team"
+        , test "assignedToDevTeam with ticket assigned to Alice from dev team" <|
+            \() ->
+                Ticket { status = New, createdBy = ( User "Jim", 3 ), assignedTo = Just (User "Alice"), comments = [] }
+                    |> assignedToDevTeam
+                    |> Expect.true "Expected ticket assigned to Alice from dev team"
+        , test "assignedToDevTeam with ticket assigned to Bob from dev team" <|
+            \() ->
+                Ticket { status = New, createdBy = ( User "James", 4 ), assignedTo = Just (User "Bob"), comments = [] }
+                    |> assignedToDevTeam
+                    |> Expect.true "Expected ticket assigned to Bob from dev team"
+        , test "assignedToDevTeam with ticket assigned to Charlie from dev team" <|
+            \() ->
+                Ticket { status = New, createdBy = ( User "Jeremy", 5 ), assignedTo = Just (User "Charlie"), comments = [] }
+                    |> assignedToDevTeam
+                    |> Expect.true "Expected ticket assigned to Charlie from dev team"
+        ]
+
+
+task4 : Test
+task4 =
+    describe "Task 4: TicketsPlease.assignTicketTo"
```

Comment:
Some of these tests are quite long, and could be made shorter and more expresser with some helper functions I think.

Something along the lines of this, which removes the irrelevant information and highlights the information that is important to the test.

```
test "assign new, unassigned ticket to Roy" <|
            \() ->
                { vanillaTicket : createdBy = user "John" }
                    |> assignTicketTo (User "Roy")
                    |> Expect.equal
                       { vanillaTicket : createdBy = user "John" }
                            , assignedTo = user "Roy"
                        )
```

Reply:
That's a good idea. I had helper functions at the beginning but then remembered that concept exercises tests are not shown, so tests should be as self contained as possible. Your approach however, is much better than mine :)